operator/src/lgen/job.rs

```diff
@@ -0,0 +1,183 @@
+}
+
+/// Create a job spec for the load generator.
+pub fn job_spec(config: JobConfig) -> JobSpec {
+    let mut env_vars = vec![
+        EnvVar {
+            name: "RUNNER_OTLP_ENDPOINT".to_owned(),
+            value: Some("http://otel:4317".to_owned()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "RUST_LOG".to_owned(),
+            value: Some("info,keramik_runner=trace".to_owned()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "GENERATOR_NAME".to_owned(),
+            value: Some(config.name.to_owned()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "GENERATOR_SCENARIO".to_owned(),
+            value: Some(config.scenario.to_owned()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "GENERATOR_TASKS".to_owned(),
+            value: Some(config.tasks.to_owned().to_string()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "GENERATOR_RUN_TIME".to_owned(),
+            value: Some(config.run_time.to_owned().to_string()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "GENERATOR_NONCE".to_owned(),
+            value: Some(config.nonce.to_owned().to_string()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "GENERATOR_PEERS_PATH".to_owned(),
+            value: Some("/keramik-peers/peers.json".to_owned()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "DID_KEY".to_owned(),
+            value: Some("did:key:z6Mkqn5jbycThHcBtakJZ8fHBQ2oVRQhXQEdQk5ZK2NDtNZA".to_owned()),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "DID_PRIVATE_KEY".to_owned(),
+            value: Some(
+                "86dce513cf0a37d4acd6d2c2e00fe4b95e0e655ca51e1a890808f5fa6f4fe65a".to_owned(),
+            ),
+            ..Default::default()
+        },
+        EnvVar {
+            name: "CERAMIC_ADMIN_PRIVATE_KEY".to_owned(),
+            value_from: Some(EnvVarSource {
+                secret_key_ref: Some(SecretKeySelector {
+                    key: "private-key".to_owned(),
+                    name: Some("ceramic-admin".to_owned()),
+                    ..Default::default()
+                }),
+                ..Default::default()
+            }),
+            ..Default::default()
+        },
+    ];
+
+    if let Some(throttle_requests) = config.throttle_requests {
+        env_vars.push(EnvVar {
+            name: "GENERATOR_THROTTLE_REQUESTS".to_owned(),
+            value: Some(throttle_requests.to_string()),
+            ..Default::default()
+        })
+    }
+
+    JobSpec {
+        backoff_limit: Some(1),
+        template: PodTemplateSpec {
+            metadata: Some(ObjectMeta {
+                labels: Some(BTreeMap::from_iter(vec![(
+                    "name".to_owned(),
+                    "load-gen-job".to_owned(),
```

Comment:
should we use the `LOAD_GENERATOR_JOB_NAME` const for name in this file as well?